File: Lecture5Code/GCD.py
#!/usr/bin/env python

##  GCD.py

import sys
if len(sys.argv) != 3:
    sys.exit("\nUsage:   %s  <integer>  <integer>\n" % sys.argv[0])

a,b = int(sys.argv[1]),int(sys.argv[2])
    
while b:                                             
    a,b = b, a%b

print("\nGCD: %d\n" % a)

File: Lecture5Code/GCD.pl
#!/usr/bin/env perl

##  GCD.pl

use strict;
use warnings;

die "\nUsage:   $0  <integer>  <integer>\n" unless @ARGV == 2;

die "At least one of your numbers is too large! Use GCDWithBigInt.pl instead\n"
    if ($ARGV[0] > 0x7f_ff_ff_ff) or ($ARGV[1] > 0x7f_ff_ff_ff);

my ($a,$b) = @ARGV;
while ($b) {
    ($a,$b) = ($b, $a % $b);
}
print "\nGCD: $a\n\n";


